internal/provider/models/processors/test/data_profiler_test.go

```diff
@@ -0,0 +1,306 @@
+  						line_fields = [".line", ".message"]
+					}`,
+				ExpectError: regexp.MustCompile("The argument \"label_fields\" is required"),
+			},
+
+			// Error: `app_fields` validates min number of items
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_data_profiler_processor" "my_processor" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						app_fields = []
+  						host_fields = [".host", ".hostname"]
+  						level_fields = [".level", ".log_level"]
+  						line_fields = [".line", ".message"]
+  						label_fields = [".labels"]
+					}`,
+				ExpectError: regexp.MustCompile("Attribute app_fields list must contain at least 1 elements, got: 0"),
+			},
+
+			// Error: `app_fields` validates max number of items
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_data_profiler_processor" "my_processor" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						app_fields = [".a", ".b", ".c", ".d", ".e", ".f", ".g", ".h", ".i", ".j", ".k"]
+  						host_fields = [".host", ".hostname"]
+  						level_fields = [".level", ".log_level"]
+  						line_fields = [".line", ".message"]
+  						label_fields = [".labels"]
+					}`,
+				ExpectError: regexp.MustCompile("Attribute app_fields list must contain at most 10 elements, got: 11"),
+			},
+
+			// Error: `app_fields` validates min item length
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_data_profiler_processor" "my_processor" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						app_fields = [""]
+  						host_fields = [".host", ".hostname"]
+  						level_fields = [".level", ".log_level"]
+  						line_fields = [".line", ".message"]
+  						label_fields = [".labels"]
+					}`,
+				ExpectError: regexp.MustCompile("Attribute app_fields\\[0\\] string length must be at least 1, got: 0"),
+			},
+
+			// Error: `app_fields` validates max item length
+			{
+				Config: GetCachedConfig(cacheKey) + fmt.Sprintf(`
+					resource "mezmo_data_profiler_processor" "my_processor" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						app_fields = [".%s"]
+  						host_fields = [".host", ".hostname"]
+  						level_fields = [".level", ".log_level"]
+  						line_fields = [".line", ".message"]
+  						label_fields = [".labels"]
+					}`, sb.String()),
+				ExpectError: regexp.MustCompile("Attribute app_fields\\[0\\] string length must be at most 512, got: 513"),
+			},
+
+			// Error: `app_fields` validates max item length
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_data_profiler_processor" "my_processor" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						app_fields = ["not-valid-field-reference"]
+  						host_fields = [".host", ".hostname"]
+  						level_fields = [".level", ".log_level"]
+  						line_fields = [".line", ".message"]
+  						label_fields = [".labels"]
+					}`,
+				ExpectError: regexp.MustCompile("Must be a valid data access syntax"),
```

Comment:
This validation is checked on the API side right?

Reply:
yeah this is hitting the API spun up in the compose